tests/libgit2/online/badssl.c

```diff
@@ -73,3 +73,24 @@ void test_online_badssl__old_cipher(void)
 	cl_git_fail(git_clone(&g_repo, "https://rc4.badssl.com/fake.git", "./fake", NULL));
 	cl_git_fail(git_clone(&g_repo, "https://rc4.badssl.com/fake.git", "./fake", &opts));
 }
+
+void test_online_badssl__untrusted(void)
+{
+	if (!g_has_ssl)
+		cl_skip();
+
+	cl_git_fail_with(
+	        GIT_ECERTIFICATE,
+	        git_clone(
+	                &g_repo, "https://untrusted-root.badssl.com/fake.git",
+	                "./fake", NULL));
+	cl_assert_equal_i(git_error_last()->klass, GIT_ERROR_SSL);
```

Comment:
With WinHTTP this error code would be GIT_ERROR_HTTP, and as pointed out the message strings are very different (I should do this stuff in the morning...).

This test is a bit pointless I suppose as it's just testing error message concatenation. What I really want to test is this logic that I added, but I'm not sure how:

```
if ((st->cert_chain->TrustStatus.dwErrorStatus &
	     ~(DWORD)(CERT_TRUST_REVOCATION_STATUS_UNKNOWN |
	              CERT_TRUST_IS_OFFLINE_REVOCATION)) != CERT_TRUST_NO_ERROR)
```